### ISSUE TITLE
Type annotations making their way into generated module with error message

### ISSUE THREAD

**Issue by leafyshark** [tajo/ladle#306]
Each time I make a few changes to the ladle stories I get a completely blank screen when visiting http://localhost:61000/ (see below)

<img width="1728" alt="Screenshot 2023-01-06 at 20 55 17" src="https://user-images.githubusercontent.com/27443324/211098386-d835d36d-3ec9-4808-8a6e-0d1fa413b3c6.png">

**Comment by its-monotype** [comment]
I'm facing this issue too. Just change smth in the code and I get a blank page, idk how to reproduce that because it can happen randomly.

**Comment by tajo** [comment]
Based on the screenshot/console, something goes wrong when refreshing/recreating a list of stories. It could be some bad escaping since that list also includes source code of each story.

A repro would be great.

**Comment by EZXOFVHeA** [comment]
I've been seeing this too. This is happening because type annotations making their way into generated module with error message.

![image](https://user-images.githubusercontent.com/100410177/212549547-d488bcaa-fb4c-4955-8250-b06edc4ebfec.png)

Would be nice if issue title could be updated to reflect the actual problem.

**Comment by leafyshark** [comment]
@EZXOFVHeA please could you elaborate on the problem and offer up a potential solution?

**Comment by some-dudes-work-account** [comment]
@leafyshark sorry, just my profile name to something more sane. 

I don't know enough about how Ladle works on the inside, so no, I can't offer a solution, but I'd imagine it would be either generating javascript instead of typescript or transpiling generated code before sending it to the browser.

**Comment by some-dudes-work-account** [comment]
Looked into it a bit more, but I give up. I have little experience with Rollup and even less with Vite, so I don't know what makes it determine when source code needs to be interpreted as Typescript and when as JS. Maybe it's based on the filename, path or perhaps virtual files are handled differently and it was the switch to `"virtual:generated-list"` as a module ID in #144 that broke it. This also aligns with Rollup's documentation somewhat:

> `--stdin=ext`
> Specify a virtual file extension when reading content from stdin. By default, Rollup will use the virtual file name - without an extension for content read from stdin. **Some plugins, however, rely on file extensions to determine if they should process a file.** See also [Reading a file from stdin](https://rollupjs.org/guide/en/#reading-a-file-from-stdin).

Any other generated code within the vite plugin doesn't seem to have any TS-specific syntax so perhaps it's a mistake and generated code was never meant to be a valid Typescript code, at least in a vite version, but once again I couldn't find a similar code in the Snowpack version. Perhaps it was added for testing purposes, I can only guess. 

@tajo probably knows more since they were the once who authored all those changes.

-----------

**tl;dr:** when there's a compilation error internal vite plugin generates TS code which makes to the browser as is, untranspiled, where it causes syntax error at runtime, preventing any of Ladle's UI from rendering and leaving users with the blank page. Supposedly got broken in #23 or #144

**Comment by its-monotype** [comment]
https://github.com/tajo/ladle/blob/78806861070e894f539cad3fa6fa913275472291/packages/ladle/lib/cli/vite-plugin/vite-plugin.js#L26

**Comment by tajo** [comment]
Quick fix in #333. 

However, the behavior is still not very user-friendly, so I created a task to improve it  https://github.com/tajo/ladle/issues/334
